### PR TITLE
optionally include descriptions in schema

### PR DESCRIPTION
**Pull request by puertomontt**
# Description
Add an option to exclude descriptions from generated openapi schema. This option existed for the Cue based generation ([pr](https://github.com/solo-io/skv2/pull/334/files)) but not the protoc-gen-openapi path. This can be used to reduce the size of the crds.

PR where this functionality is used: https://github.com/solo-io/gloo-mesh-enterprise/pull/18641. You can see that the descriptions were removed from the CRD.

# Context

The GME CRD chart is massive and approaching the limit that can be stored in a k8s secret. Removing the descriptions reduces the size drastically.

BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh-enterprise/issues/18562

### CODE REVIEW

File: codegen/collector/executor.go [hunk 1]
```diff
@@ -108,11 +108,12 @@ func (o *OpenApiProtocExecutor) Execute(protoFile string, toFile string, imports
 	_, fileName := filepath.Split(protoFile)
 	directoryName := fileName[0 : len(fileName)-len(filepath.Ext(fileName))]
 
-	baseArgument := "--openapi_out=yaml=true,single_file=false,include_description=true,multiline_description=true,proto_oneof=true,int_native=true"
+	baseArgument := "--openapi_out=yaml=true,single_file=false,multiline_description=true,proto_oneof=true,int_native=true"
```
Comment (jenshu) [inline codegen/collector/executor.go:111]: is there a test suite for this?
or can you link to a PR that shows the GME CRDs diff after this change?

**Review by jenshu — APPROVED** [review]

File: codegen/collector/executor.go (hunk 1 above)
Comment (puertomontt) [reply codegen/collector/executor.go:111]: https://github.com/solo-io/gloo-mesh-enterprise/pull/18641

File: codegen/proto/schemagen/protoc.go [hunk 2]
```diff
@@ -19,12 +19,14 @@ import (
 
 // Implementation of JsonSchemaGenerator that uses a plugin for the protocol buffer compiler
 type protocGenerator struct {
-	validationSchemaOptions *ValidationSchemaOptions
+	validationSchemaOptions     *ValidationSchemaOptions
+	excludeDescriptionsInSchema bool
```
[the remark this answers is not in the document]
Comment (puertomontt) [reply codegen/proto/schemagen/protoc.go:23]: It's not related to validation schema specifically, which is why I felt it should be separate

[the remark this answers is not in the document]
Comment (puertomontt) [reply codegen/proto/schemagen/protoc.go:23]: open to suggestions on how to refactor though.
Maybe a high level options with validationSchemaOpts nested along with the excludeDescriptionsInSchema?

[the remark this answers is not in the document]
Comment (jenshu) [reply codegen/proto/schemagen/protoc.go:23]: IMO the skip descriptions flag does sound like it's related to the validation schema options, as they all go into the openapi schema in the CRD. but i could be mistaken as i'm not too familiar with the usage of these options

**Comment by puertomontt** [comment]
Going to park this for now. We might try having links to the docs site rather than removing them

**Pull request closed without merging**